### ISSUE TITLE
[Jepsen] [YSQL] Occasional long recovery during partition-one 

### ISSUE THREAD

**Issue by frozenspider** [yugabyte/yugabyte-db#1992]
(Moved from #950)

`partition-one` nemesis may knock cluster unconscious for ~100 seconds. Then it recovers (even though partition is still in action)

Comment by @aphyr (https://github.com/YugaByte/yugabyte-db/issues/950#issuecomment-518664618):

> Just a heads-up that there may be more work to do here: I've observed occasional ~100-second recovery times for a partition isolating a single node, on version 1.3.1.0. For instance: [20190805T185957.000-0400.zip](https://github.com/YugaByte/yugabyte-db/files/3472420/20190805T185957.000-0400.zip).
> 
> ![latency-raw (32)](https://user-images.githubusercontent.com/3748/62542929-d13bb880-b82a-11e9-829c-d7a70c11c10f.png)

Command-line suggested by @aphyr was 
```
lein run test --os debian --version 1.3.1.0 --nemesis partition-one --nemesis-interval 200 --nemesis-schedule fixed --time-limit 500 -w ysql/append --test-count 10
```
on commit `ed8ecfb9146816cb99e458c9cb92f59e19a7b78f`

**Comment by frozenspider** [comment]
Was able to reproduce that (once in 20 attempts):
[logs-jepsen_2019-08-08_00-21-01_append_n-part1_t500.zip](https://github.com/YugaByte/yugabyte-db/files/3478591/logs-jepsen_2019-08-08_00-21-01_append_n-part1_t500.zip)
![latency-raw](https://user-images.githubusercontent.com/2077017/62651971-a90ba100-b973-11e9-9cd6-95961675a38f.png)

**Comment by frozenspider** [comment]
This still occurrs on commit 9f4448d (included in v2.1.3), albeit extremely rarely (1 in 459 runs)
![latency-raw](https://user-images.githubusercontent.com/2077017/82816416-8d4a9a80-9eb4-11ea-8476-6d13ec674f65.png)
Logs: [20200522T204235.000Z.tar.gz](https://github.com/yugabyte/yugabyte-db/files/4677361/20200522T204235.000Z.tar.gz)

**Comment by qvad** [comment]
Note that after this fix, the recovery window for YSQL should be down to ~30s for releases 2.13.1 and beyond.
For older releases, i.e. 2.12 and older, the recovery window will be O ( number of nodes downed) ~= 15s * (num_nodes downed + 1).

See https://github.com/yugabyte/yugabyte-db/issues/11799#issuecomment-1098506726
